src/app/teams/join/[teamId]/page.tsx

```diff
@@ -0,0 +1,32 @@
+import {getServerAuthSession} from "~/server/auth";
+import {api} from "~/trpc/server";
+import JoinTeam from "~/app/_components/team/joinTeam";
+
+export default async function Home({ params }: { params: { teamId: string } }) {
+  const { teamId } = params;
+  const session = await getServerAuthSession();
+
+  if (!session) {
```

Comment:
Great opportunity to [pick up these gold nuggets](https://www.youtube.com/watch?v=qiC3hGeff5c&ab_channel=DevtoolsTech) and put them to use.